src/token/TokenUpgrade.sol

```diff
@@ -59,6 +60,7 @@ contract TokenUpgrade is AccessControl {
         if (address(tokenIn_) == address(tokenOut_)) revert SameToken(address(tokenIn_));
         if (address(tokensIn[tokenIn_].reverse) != address(0)) revert TokenInAlreadyRegistered(address(tokenIn_));
         if (address(tokensOut[tokenOut_].reverse) != address(0)) revert TokenOutAlreadyRegistered(address(tokenOut_));
+        if (tokenOut_.balanceOf(address(this)) != tokenOut_.totalSupply()) revert TotalSupplyNotPresent();
```

Comment:
It is not required for the TokenUpgrade to have the complete supply of `tokenOut`.

There are several valid scenarios in which there might be `TokenOut` tokens owned by other users, and scenarios in which we can't restrict the minting of `TokenOut`.